reo/process_results.py

```diff
@@ -234,6 +510,72 @@ def upfront_capex_after_incentives(self):
                                               (1 - self.inputs["Financial"]["offtaker_tax_pct"])
             return round(upfront_capex_after_incentives, 2)
 
+        def calculate_lcoe(self, tech_results_dict, tech_inputs_dict, financials):
+            """
+            The LCOE is calculated as annualized costs (capital and O+M translated to current value) divided by annualized energy
+            output
+            :param tech_results_dict: dict of model results (i.e. outputs from PVModel )
+            :param tech_inputs_dict: dict of model results (i.e. inputs to PVModel )
+            :param financials: financial model storing input financial parameters
+            :return: float, LCOE in US dollars per kWh
+            """
+            years = financials.analysis_years # length of financial life
+
+            if financials.two_party_ownership:
+                discount_pct = financials.owner_discount_pct
+            else:
+                discount_pct = financials.offtaker_discount_pct
+
+            new_kw = (tech_results_dict.get('size_kw') or 0) - (tech_inputs_dict.get('existing_kw') or 0) # new capacity
+            
+            if new_kw == 0:
+                return None
+
+            capital_costs = new_kw * tech_inputs_dict['installed_cost_us_dollars_per_kw'] # pre-incentive capital costs
+            
+            annual_om = new_kw * tech_inputs_dict['om_cost_us_dollars_per_kw'] # NPV of O&M charges escalated over financial life
```

Comment:
should we not include tax benefits in the LCOE calculation? For example the O&M costs will be less by the tax rate if the site pays taxes.

Reply:
@NLaws good catch here! talked with Dan, he hadn't been considering tax benefits (and MACRS, federal ITC) when we spoke before about how to calculate LCOE. He's looking into it now and will have some advice on how to include these soon